app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php

```diff
@@ -21,16 +23,33 @@ class Stock extends UnsubscribeController
     protected $productRepository;
 
     /**
-     * @param \Magento\Framework\App\Action\Context $context
-     * @param \Magento\Customer\Model\Session $customerSession
-     * @param \Magento\Catalog\Api\ProductRepositoryInterface $productRepository
+     * @var Magento\Store\Model\StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var \Magento\ProductAlert\Model\StockFactory
+     */
+    protected $stockFactory;
+
+    /**
+     * Stock constructor.
+     * @param Context $context
+     * @param CustomerSession $customerSession
+     * @param ProductRepositoryInterface $productRepository
+     * @param StoreManagerInterface $storeManager
+     * @param \Magento\ProductAlert\Model\StockFactory $stockFactory
      */
     public function __construct(
         Context $context,
         CustomerSession $customerSession,
-        ProductRepositoryInterface $productRepository
+        ProductRepositoryInterface $productRepository,
+        StoreManagerInterface $storeManager,
+        \Magento\ProductAlert\Model\StockFactory $stockFactory
```

Comment:
Please add default value `null` to keep this changes Backward compatible.
Please use short name of the class.

Reply:
@swnsma
done

Reply:
I suppose, `\Magento\ProductAlert\Model\StockFactory` is not the same as `Magento\ProductAlert\Controller\Unsubscribe\Stock`, is not it?
In result we will have 2 short names: `StockFactory` and `Stock`